contributing/core/developing-using-local-app.md

```diff
@@ -0,0 +1,70 @@
+# Developing Using Your Local Version of Next.js
+
+There are two options to develop with your local version of the codebase:
+
+### Set as a local dependency in package.json
+
+1. In your app's `package.json`, replace:
+
+   ```json
+   "next": "<next-version>",
+   ```
+
+   with:
+
+   ```json
+   "next": "file:/path/to/next.js/packages/next",
+   ```
+
+2. In your app's root directory, make sure to remove `next` from `node_modules` with:
+
+   ```sh
+   rm -rf ./node_modules/next
+   ```
+
+3. In your app's root directory, run:
+
+   ```sh
+   pnpm i
+   ```
+
+   to re-install all of the dependencies.
+
+   Note that Next will be copied from the locally compiled version as opposed to from being downloaded from the NPM registry.
+
+4. Run your application as you normally would.
+
+5. To update your app's dependencies, after you've made changes to your local `next` repository. In your app's root directory, run:
+
+   ```sh
+   pnpm install --force
+   ```
+
+#### Troubleshooting
+
+- If you see the below error while running `pnpm dev` with next:
+
+```
+Failed to load SWC binary, see more info here: https://nextjs.org/docs/messages/failed-loading-swc
+```
+
+Try to add the below section to your `package.json`, then run again
+
+```json
+"optionalDependencies": {
+  "@next/swc-linux-x64-gnu": "canary",
+  "@next/swc-win32-x64-msvc": "canary",
+  "@next/swc-darwin-x64": "canary",
+  "@next/swc-darwin-arm64": "canary"
+},
+```
+
+### Develop inside the monorepo
```

Comment:
Is it worth to mention you can create a dir in the monorepo without a `package.json` and just run `pnpm next ./my-dev-project`?

Reply:
https://github.com/vercel/next.js/pull/40515/commits/89afdef57a2760aff4c8ce554212c6d36fa61a26